hypercall.h

```diff
@@ -10,11 +10,12 @@ int strnlen_n(char* s, int max_len){
 
 // #if defined(__x86_64__)
```

Comment:
Unrelated to your changes here, this hypercall implementation seems like it could be improved.

I like how you added a second argument, that makes it way more usable since you can then do like an action enum combined with a data field. If you have two arguments, I'd highly recommend dropping the magic value and just checking your action enum against a set of known magic values (e.g., nvram_get can be 0x123456 and nvram-set could be 0x654321).

Some of these seem to be restoring register state after the hypercall in the inline asm which should be unnecessary since the compiler is told that the asm clobbers some instructions. For example, check out the difference in the arm and mips hypercalls here: https://github.com/panda-re/linux/blob/fd4.10_hc/include/linux/hypercall.h - I think removing some of the redundant pops should get you a speed boost since the compiler will be able to optimize the state restoration and general use of registers.

Reply:
I'm not sure I agree with this, the magic value check decides who the hypercall "dispatches" to, while the command determines behavior within the specific hypercall handler. 

That is to say, if I want to make a second system which communicates with another pypanda plugin, how would that work?

Reply:
Every panda plugin is informed of every hypercall, right? So I don't think we need to worry about dispatching - so long as two plugins don't use overlapping magic values, they'll be able work independently? A guest triggers a hypercall with some magic enum value in one arg and some data in the 2nd arg. Multiple plugins check if the first value is one they're built to handle and the one that does handle it consumes the 2nd arg and does something.

@jamcleod do you think we should build an abstraction layer on top of the panda hypercall callback that provides some kind of PPP-based callback where plugins register magic values with it and are then only notified on "their" hypercalls versus any?

I do think hypercall.h should be neutral to this sort of thing, and perhaps ultimately merged into the main panda repo, though I think that if we were to do that we should allow callers to set more arguments than just 1 or 2.

Reply:
So I did argv and argc because I figured vaargs would be extermely problematic. The problem with generalizing this version of hypercall is because  strings and paging wouldn't work well with binary data like in [#52](https://github.com/panda-re/igloo/issues/52) unless you wanted to base64 encode. I don't know how much you use hypercalls and whether plugin specific  magic values makes more sense than command magic values.